Add caveat about Symbol.species being not recommended, new SAB[@@species] page (#20601)
mdn/content: files/en-us/web/javascript/reference/global_objects/map/@@species/index.md

{{JSRef}}

The **`Map[@@species]`** accessor property is an unused accessor property specifying how to copy `Map` objects.

## Syntax

```js
Map[Symbol.species]
```

### Return value

The value of the constructor (`this`) on which `get @@species` was called. The return value is used to construct copied `Map` instances.

## Description

The `@@species` accessor property returns the default constructor for `Map` objects. Subclass constructors may override it to change the constructor assignment.

> **Note:** This property is currently unused by all `Map` methods.

## Examples

### Species in ordinary objects

The `@@species` property returns the default constructor function, which is the `Map` constructor for `Map`.

```js
Map[Symbol.species]; // function Map()
```

### Species in derived objects

> In a derived collection object (e.g. your custom map `MyMap`), the `MyMap` species is the `MyMap` constructor. However, you might want to overwrite this, in order to return parent `Map` objects in your derived class methods:

Reviewer: ```suggestion
In a derived collection object such as the custom map `MyMap`, the `MyMap` species is the `MyMap` constructor. However, you might want to overwrite this, in order to return parent `Map` objects in your derived class methods:
```
